feat: add loading indicator when changing password and saving profile

techno-disaster opened the pull request:
### Description
Add loading indicator on screen to help user understand what's happening.
If a user has a slow internet connection, the changing password request can take long and without a loading indicator he/she might think the apps stuck. Loading indicators will tell that the process is currently going on  
Fixes #67
Fixes #94

### Flutter Channel:
- [x] I have used the Flutter Beta channel on my local machine 

### Type of Change:
**Delete irrelevant options.**

- User Interface

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)



### How Has This Been Tested?

Physical device. Check gif below
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/52817235/82616814-e8963780-9beb-11ea-90ce-27e8a07d1e27.gif)


### Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings

Commit:
feat: add loading indicator when changing password & saving profile

Commit:
cleanup

Commit:
disable interrupting process loading indicator

Techno-Disaster commented:
@anitab-org/qa-team can someone test this? 
Steps : 
Change password, a loading indicator like in the gif should load up, until the snackbar comes up
Change something in Profile and save profile, a loading indicator should show up until the snacbar comes

bartekpacia commented on lib/screens/home/pages/profile/profile_page.dart:
```diff
@@ -220,4 +222,12 @@ class _ProfilePageState extends State<ProfilePage> {
     );
   }
+
+  Future<void> showloader(BuildContext context) {
```
Please change the name to _showProgressIndicator()

techno-disaster replied:
Done

bartekpacia commented on lib/screens/settings/settings_screen.dart:
```diff
@@ -74,16 +76,15 @@ class SettingsScreen extends StatelessWidget {
   }
 
-  Future<void> _showChangePasswordDialog(BuildContext context) async {
+  Future<void> _showChangePasswordDialog(BuildContext theContext) async {
```
If we want to name `BuildContext`s like this, let's use `topContext` instead of `theContext`.

techno-disaster replied:
sure. will change asap

techno-disaster replied:
Done

bartekpacia commented:
Great job overall👍

bartekpacia reviewed (dismissed):
Looks good.
EDIT: Looks I had a second reviewing screen opened in the second browser tab🤦‍♀️

Techno-Disaster commented:
> Looks good.

Still have to make some changes.

bartekpacia committed:
Merge branch 'develop' into issue-67-loading-indicators

Commit:
cleaner code

Techno-Disaster commented:
#### Found one more page which can use loader and cleaned he code a bit

![ezgif com-video-to-gif(1)](https://user-images.githubusercontent.com/52817235/83078724-d9096980-a097-11ea-9276-62b71d3367d8.gif)

Techno-Disaster commented:
I had to remove ```final``` from the bloc because I wanted to set the message to null, after we show the snackbar. without this the loader runs fine for the first time, but if the user hits send request again, then as state.message is not null anymore it wont enter the ifcase, and no snackbar will be shown + continuous loading.

bartekpacia reviewed (dismissed):
LGTM

Techno-Disaster commented:
@isabelcosta can you review this too? 👀 Thanks

robotjellyzone reviewed (dismissed):
Looks good to me @Techno-Disaster ! just want to know why did you downgraded some of the versions in pubspec file ?

Techno-Disaster commented:
> Looks good to me @Techno-Disaster ! just want to know why did you downgraded some of the versions in pubspec file ?

The pubspec.lock is an auto generated file. We store all our dependencies in pubspec.yaml file.

Techno-Disaster commented:
Working on #94  (It needs loading indicators as well). Lets review it after that's pushed too

Commit:
Merge branch 'develop' into issue-67-loading-indicators

Commit:
add indicator to tasks page

Techno-Disaster commented:
Added indicators to tasks page.
![ezgif com-video-to-gif(1)](https://user-images.githubusercontent.com/52817235/84447567-f593cd00-ac65-11ea-8fb9-8d12da71e52e.gif)

Techno-Disaster commented:
@anitab-org/qa-team  can someone give this a quick test? 
Thank You

Gaurav192 commented on lib/screens/send_request/bloc/send_request_state.dart:
```diff
@@ -1,7 +1,7 @@
 
 abstract class SendRequestState extends Equatable {
-  final String message;
+  String message;
```
Removing final is not a proper solution.
I would suggest to use a resetEvent which yields initalState once message has been listened to.

techno-disaster replied:
@Gaurav192 thanks for the suggestion, updated the pr can you give it a quick look? Thanks

Commit:
reset snackbar message

Gaurav192 commented on lib/screens/send_request/bloc/send_request_bloc.dart:
```diff
@@ -28,5 +28,13 @@ class SendRequestBloc extends Bloc<SendRequestEvent, SendRequestState> {
       }
     }
+    if (event is ResetSnackbarMessage) {
+      try {
```
try catch is not required here.
yield initialState;  is enough

techno-disaster replied:
@Gaurav192 done :)

Commit:
CleanUp

HaripriyaB commented on lib/screens/home/pages/relation/relation_page.dart:
```diff
@@ -163,8 +165,8 @@ class _RelationPageState extends State<RelationPage> {
                     GestureDetector(
                       onTap: () {
                         if (!task.isDone) {
-                          context.toast("hey");
```
@Techno-Disaster I think taking off this line was supposed to be for another issue which we discussed in last meeting right?

techno-disaster replied:
@HaripriyaB yes, but IMHO it's just 1 small line and it doesn't really make any sense to have a whole new Issue for it.

HaripriyaB replied:
yes that makes sense...

bartekpacia reviewed (dismissed):
nice!

robotjellyzone commented:
> @anitab-org/qa-team can someone give this a quick test?
> Thank You

I have tested this PR @Techno-Disaster as per your request . The needed feature is working as can be seen below :-
![testing](https://user-images.githubusercontent.com/36916536/84567176-88cb2080-ad94-11ea-852d-1d42c20802d3.gif)

robotjellyzone commented:
But  I don't know why relation page isn't working as it can be seen from the following gif that all other pages are working but i am unable to see the relation page as its causing the app to crash ... with the following message on logcat  -->

 `INFO: 2020-06-13 16:38:29.914830: Transition { currentState: RelationPageLoading, event: RelationPageShowed, nextState: RelationPageFailure }`

![relation-page-not-working](https://user-images.githubusercontent.com/36916536/84567272-258dbe00-ad95-11ea-9844-20ee946e8d6c.gif)

robotjellyzone commented:
May be the changes done in the relation page needs to be reviewed again 🤔 @Techno-Disaster ?

robotjellyzone reviewed (commented):
May be the changes done in the relation page needs to be reviewed again 🤔 @Techno-Disaster ?

Commit:
fix relation screen when not in a relation and add signUp page loader

Techno-Disaster commented:
@robotjellyzone thanks for reporting this, this happened when the user was not in a relationship, the issue is now fixed. Can you test again? i also added a loading indicator to the signUp page.

Techno-Disaster commented:
@anitab-org/qa-team can anyone give this a quick test again? Thank You

robotjellyzone reviewed (dismissed):
Hi @Techno-Disaster i have tested the required feature & the relation page , everything is now working fine 

great  👍 

![testing_pr_2](https://user-images.githubusercontent.com/36916536/84742545-36754400-afce-11ea-8cca-45b6713110fc.gif)

bartekpacia commented on lib/widgets/loading_indicator.dart:
```diff
@@ -9,3 +9,11 @@ class LoadingIndicator extends StatelessWidget {
   }
 }
+
+Future<void> showProgressIndicator(BuildContext context) {
```
`topcontext` -> `topContext` ;)

bartekpacia replied:
I marked wrong lines, but you surely understand what I mean

techno-disaster replied:
got it, will get it done asap :)

robotjellyzone replied:
> `topcontext` -> `topContext` ;)

camel casing good catch 👍

techno-disaster replied:
Done

bartekpacia reviewed (dismissed):
Nice job :)

Commit:
smol fix

Techno-Disaster commented:
@anitab-org/mentorship-flutter-maintainers Can you take a quick look at this? The PR has been tested (all OK) just needs a quick review now

isabelcosta reviewed (approved):
Knowing very little about flutter, I am approving because this functionality seems to work.
@Techno-Disaster great work.
I am counting on other reviewers for code quality check. @bartekpacia you have to re-review after recent code changes.

Techno-Disaster commented:
@bartekpacia  could you check this again? previous review git dismissed due to the ```topcontext``` > ```topContext``` change

bartekpacia reviewed (approved):
thanks for using the `camelCase`, good job!
